Z64Utils/Z64/Z64ObjectAnalyzer.cs

```diff
@@ -447,6 +496,343 @@ public static void FindAnimations(Z64Object obj, byte[] data, int segmentId)
+                    Func<SegmentedAddress, int, int, bool, bool> validSegmentAddr = (seg, num, align, allowNull) =>
+                        ((seg.SegmentId == segmentId && seg.SegmentOff < data.Length &&
+                          obj.IsOffsetFree((int)seg.SegmentOff) && seg.SegmentOff % align == 0) ||
+                         (allowNull && num == 0 && seg.VAddr == 0));
+
+                    if (validSegmentAddr(verticesSeg, nVertices, 2, false) &&
+                        validSegmentAddr(polygonsSeg, nPolygons, 2, false) &&
+                        validSegmentAddr(surfaceTypesSeg, surfaceTypesSize / Z64Object.CollisionSurfaceTypesHolder.ENTRY_SIZE, 4, false) &&
+                        validSegmentAddr(camDataSeg, camDataSize / Z64Object.CollisionCamDataHolder.ENTRY_SIZE, 2, true) &&
+                        validSegmentAddr(waterBoxesSeg, nWaterBoxes, 4, true))
+                    {
+                        // check padding between surface types and polygons
+                        if (surfaceTypesSize != 0 && surfaceTypesPadLen != 0)
+                        {
+                            byte[] possiblePadding = new byte[surfaceTypesPadLen];
+
+                            // 'Offset and length were out of bounds for the array or count is greater than the number of
+                            // elements from index to the end of the source collection.'
+                            Buffer.BlockCopy(data, (int)surfaceTypesSeg.SegmentOff + surfaceTypesSize, possiblePadding,
+                                0, surfaceTypesPadLen);
+                            // if assumed struct padding is nonzero, consider invalid
+                            if (possiblePadding.Any(b => b != 0))
+                                continue;
+                        }
+                        // check padding between cam data and surface types
+                        if (camDataSeg.VAddr != 0 && camDataSize != 0 && camDataPadLen != 0)
+                        {
+                            byte[] possiblePadding = new byte[camDataPadLen];
+                            Buffer.BlockCopy(data, (int)surfaceTypesSeg.SegmentOff + camDataSize, possiblePadding,
+                                0, camDataPadLen);
+                            // if assumed struct padding is nonzero, consider invalid
+                            if (possiblePadding.Any(b => b != 0))
+                                continue;
+                        }
+
+                        Z64Object.ColHeaderHolder colHeader = obj.AddCollisionHeader(off: i);
+
+                        colHeader.VerticesHolder = obj.AddCollisionVertices(nVertices, off: (int)verticesSeg.SegmentOff);
+                        colHeader.PolygonsHolder = obj.AddCollisionPolygons(nPolygons, off: (int)polygonsSeg.SegmentOff);
+                        colHeader.SurfaceTypesHolder = obj.AddCollisionSurfaceTypes(

```

Comment:
I agree this would be more appropriate since as you say the current way doesn't work for importing. This is a more general issue that applies to other holders already in the repo though, so I'd like to hold off on this and fix them in another PR, which I am happy to do.

Reply:
What other holders? I only know about specifically CI textures palettes (we may want to open an issue to discuss implementing the "update holder references" thing and list relevant holders)